### DISCUSSION TITLE
v1.13

### DISCUSSION THREAD

**Discussion by dorkster** [flareteam/flare-game#898] [Announcements]
## Engine features

* Added additional requirement properties (requires\_status, requires\_item, etc) to enemy and NPC map objects. The properties now match what is available for map events.
* When 'mouse aim' is disabled, the Shift key can now be used with the direction keys to change the player's orientation without moving.
* Added optional 'fog-of-war' feature (r-a-cristian-93)
* Added support for multiple sets/loadouts of equipped items (r-a-cristian-93)
* Added random\_status attribute to Events for setting/unsetting a random status from a list.
* Reworked gamepad handling using the SDL\_GameController API.
* Several new key bindings, many of which will benefit gamepad users.
* Added an "action picker" for interacting with items/powers when using a gamepad or a touchscreen.
* Added misc.selection\_rect\_corner\_size to engine/widget\_settings.txt. This allows for a nicer "targeting reticle" appearance when selecting widgets without a mouse.
* Added 'background' property for all menus. Menus will fall back to the existing hard-coded filenames if this property is omitted.
* Added 'close', 'label\_title', and 'action\_list' properties to MenuConfirm configuration.
* Added 'render\_layers' section to StatBlock definitions. It has the same syntax and functionality as engine/hero\_layers.txt, so that file is now considered deprecated.
* Added 'animation\_slots' section to StatBlock definitions. This allows for multiple animation layers on non-player entities that don't have an inventory like the player.
* Added a 'direction' property to map NPCs, indentical to the property of the same name for map enemies.
* Added 'vendor\_ratio\_buy', 'vendor\_ratio\_sell', and 'vendor\_ratio\_sell\_old' properties to engine/loot.txt. These deprecate 'vendor\_ratio' and 'vendor\_ratio\_buyback'.
* Added 'vendor\_ratio\_buy', 'vendor\_ratio\_sell', and 'vendor\_ratio\_sell\_old' properties to NPCs. These can be used to override the global settings in engine/loot.txt.
* Added 'tooltip\_visible\_max' to engine/tootlips.txt to control the maximum number of visible floating tooltips.

## Engine fixes

* Fix enemies not being able to target NPC followers
* Fix loot and sound effects being cleared in StatBlock files with multiple sections
* Fix potential crash when removing summons that would be out-of-bounds on the next map.
* Fix bug where a damage type with no name could be removed if another damage type was defined after it.
* Don't show hidden stats under 'Related Stats' section of tooltips of Character menu.
* Fix bug where negative regen wouldn't work if the resource was 100% full.
* Fix bug where negative regen would overflow after reaching zero.
* Fix bug that caused built-in effect types to be ignored by Power post\_effect.
* Fix disappearing sprites due to render priority overflow.
* Fading of corpses is now correctly disabled when corpse\_timeout=0 is used in engine/misc.txt.
* Fix cast animation not playing when using an inventory item that is not instant.
* Fixed a bug on Android where loading the default settings would exit fullscreen mode.
* Fixed crash in NPC dialog when using gamepad.
* Fixed several small bugs with navigating UI elements with a gamepad.
* Fixed incorrect handling of gamepad stick/trigger deadzone.
* Fixed fullscreen setting being turned off when loading configuration defaults on Android.
* Fixed input hint for item tooltips when using a touchscreen.
* Improved file access handling on Android versions 10 and up. This fixes the infinite cutscene loop when starting a new game.
* Fixed memory leak when loading NPCs.
* Entity animations will now fall back to the 'die' animation when 'critdie' isn't defined.
* Improved text rendering by blending text with a copy of itself.
* Fix infinite loop that could occur when setting unlocked powers.
* Elemental resistances can now be used with 'stat', 'stat\_per\_level', and 'stat\_per\_primary' in StatBlock definitions.

## Game updates

* New formula for the 'Shield' power. Instead of Mental Damage, it is now based off a combination of player level, Mental stat, and Defense stat. This should increase its usefulness on mixed builds.
* Added 'Super' and 'Ultra' variants of Health and Mana potions. They can be obtained through the Alchemy system.
* New icons for Health and Mana potions variants.
* Fixed Alchemist's Skirt so that it works with all potions.
* Prevent the player from damaging Mez before being trapped.
* Made Lezaith immune to slow/stun effects.
* Goblin Thief can now steal all available kinds of potions. Its drop table has also been expanded to make it easier to replenish lost potions.
* Improved contrast for "highlight" state of red square buttons ("close" and "plus").
* The minimap is now gradually revealed as the player explores.
* Added icon artwork to the equipment slots on the Inventory menu.

## Translation updates

* Basque (eu) update (aitzkora)
* Scottish Gaelic (gd) update (GunChleoc)
* Norwegian Bokmal (nb) update (Elias Nykrem)
* Polish (pl) update (Pawonashi)
* Russian (ru) update (Igor Polyakov)

<hr /><em>This discussion was created from the release <a href='https://github.com/flareteam/flare-game/releases/tag/v1.13'>v1.13</a>.</em>

**Comment by WithinAmnesia** [comment]
Way to go @dorkster ! Hip hip hooray XD. Thank you for all of your hard work and dedication to making all of the Flare systems grow and improve. I hope that the creative community can live up to your dedication and make some masterpieces that we all can be inspired by and for the entire world to enjoy :-).
